Implement new_edits flag on bulk updates

robknight opened the pull request:
Will include further commits to implement new_edits flag - at the moment just has a failing test for revision IDs on bulk doc updates.

robknight committed:
Added test for revision ID list length following bulk docs update

dickolsson commented:
@robknight You probably need to `git rebase origin/8.x-1.x` in your clone, because I recently merged in some fairly big changes.

robknight committed:
Correctly handle loading of entities that have been flagged as deleted

robknight committed:
Reset the entity cache when loading the entity for comparison

robknight committed:
Compare field values, not field objects

robknight committed:
Add the 'new_edits' flag to the normalizer context array if set in the request body

robknight committed:
Test that revision IDs do not change when 'new_edits' flag is set

dickolsson commented on src/Normalizer/ContentEntityNormalizer.php:
```diff
@@ -145,13 +145,30 @@ public function denormalize($data, $class, $format = NULL, array $context = arra
+      $entity = $this->entityManager->getStorage($entity_type_id)->load($entity_id);
+      if (!$entity) {
+        $entity = $this->entityManager->getStorage($entity_type_id)->loadDeleted($entity_id);
+      }
```
Would be shorter and nicer if we did something like:

```
$storage = $this->entityManager->getStorage($entity_type_id);
$entity = $storage->load($entity_id) ?: $storage->loadDeleted($entity_id);
```

Then we could use `$storage` when we also do the create below.

robknight committed:
Cleaned up entity loading and creation code.

As per dickolsson's comment on https://github.com/dickolsson/drupal-relaxed/pull/5#discussion_r18429669

dickolsson commented on src/Normalizer/ContentEntityNormalizer.php:
```diff
@@ -145,13 +145,28 @@ public function denormalize($data, $class, $format = NULL, array $context = arra
+    }
+    else {
+      /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+      $storage->create($data);
```
The tests are failing because we still need to do `$entity = $storage->create($data)` here.

dickolsson commented on src/Normalizer/ContentEntityNormalizer.php:
```diff
@@ -145,13 +145,28 @@ public function denormalize($data, $class, $format = NULL, array $context = arra
+    $storage = $this->entityManager->getStorage($entity_type_id);
+
+    if ($entity_id) {
+      $entity = $storage->load($entity_id) ?: $storage->loadDeleted($entity_id);
```
Let's add `/** @var \Drupal\Core\Entity\ContentEntityInterface $entity */` above this line for consistency.

dickolsson commented on src/Normalizer/ContentEntityNormalizer.php:
```diff
@@ -145,13 +145,28 @@ public function denormalize($data, $class, $format = NULL, array $context = arra
+    if ($entity_id) {
+      $entity = $storage->load($entity_id) ?: $storage->loadDeleted($entity_id);
+      foreach ($data as $name => $value) {
+        $entity->$name = $value;
```
It would be better/safer to use this syntax instead: `$entity->{$name} = $value`

robknight committed:
Annotate the assignment of the  variable

robknight committed:
Use safer syntax for accessing variable field names

robknight committed:
Entity creation now correctly assigns  variable
